src/main.ts

```diff
@@ -56,36 +94,47 @@ export const LunchMoneyKrakenConnection: LunchMoneyCryptoConnection<
         key = parts[0];
       }
 
-      let type = 'crypto';
-
-      if (key.startsWith('Z')) {
-        type = 'cash';
-      }
-
-      let cleaned: string | null = null;
-      if (key.startsWith('Z') || key.startsWith('X')) {
-        // According to https://support.kraken.com/hc/en-us/articles/360001206766-Bitcoin-currency-code-XBT-vs-BTC
-        // Stripping the Z or X of the front works most of the time but there are some exceptions
-        if (key === 'XBT' || key === 'XXBT') {
-          cleaned = 'BTC';
-        } else if (key === 'XDG') {
-          cleaned = 'DOGE';
-        } else {
-          cleaned = key.substr(1);
-        }
+      // Maps some special kraken symbols to common ones.
+      let { symbol, type } = transformSymbol(key);
+      if (!symbol) {
+        symbol = key;
+        type = 'crypto';
       }
 
-      let balance = updateBalance(balances[cleaned ?? key], value);
+      // if (key.startsWith('Z')) {
+      //   type = 'cash';
+      // }
+
+      // let cleaned: string | null = null;
+      // if (key.startsWith('Z') || key.startsWith('X')) {
+      //   if (key === 'XBT' || key === 'XXBT') {
+      //     cleaned = 'BTC';
+      //   } else if (key === 'XDG' || key == 'XXDG') {
+      //     cleaned = 'DOGE';
+      //   } else {
+      //     cleaned = key.substr(1);
+      //   }
+      // }
+
+      let balance = updateBalance(balances[symbol], value);
```

Comment:
I left the commented out code, so you could see the updated logic based on your comment